src/libecalc/presentation/yaml/yaml_types/components/system/yaml_consumer_system.py

```diff
@@ -11,87 +11,33 @@
 )
 from libecalc.presentation.yaml.yaml_types.components.yaml_pump import YamlPump
-from libecalc.presentation.yaml.yaml_types.yaml_stream import YamlStream
+from libecalc.presentation.yaml.yaml_types.yaml_stream_conditions import (
+    YamlStreamConditions,
+)
 
 opt_expr_list = Optional[List[ExpressionType]]
 
 PriorityID = str
 StreamID = str
 ConsumerID = str
 
-YamlConsumerStreamConditions = Dict[StreamID, YamlStream]
+YamlConsumerStreamConditions = Dict[StreamID, YamlStreamConditions]
 YamlConsumerStreamConditionsMap = Dict[ConsumerID, YamlConsumerStreamConditions]
 YamlPriorities = Dict[PriorityID, YamlConsumerStreamConditionsMap]
 
-
-def map_consumer(
-    consumer: Union[YamlPump, YamlCompressor],
-    consumes: ConsumptionType,
-    regularity: Dict[datetime, Expression],
-    category: str,
-    target_period: Period,
-    references: References,
-    fuel: Optional[Dict[datetime, dto.types.FuelType]],
-):
-    """
-    Consumer mapper, should probably call ConsumerMapper, but need to figure out differences between consumer in system
-    and not in system. i.e. with or without stream conditions.
-    """
-    if consumer.component_type == ComponentType.COMPRESSOR_V2:
-        return dto.components.CompressorComponent(
-            consumes=consumes,
-            regularity=regularity,
-            name=consumer.name,
-            user_defined_category=define_time_model_for_period(
-                consumer.category or category, target_period=target_period
-            ),
-            fuel=fuel,
-            energy_usage_model={
-                timestep: resolve_and_validate_reference(
-                    value=reference,
-                    references=references.models,
-                )
-                for timestep, reference in define_time_model_for_period(
-                    consumer.energy_usage_model, target_period=target_period
-                ).items()
-            },
-        )
-    elif consumer.component_type == ComponentType.PUMP_V2:
-        return dto.components.PumpComponent(
-            consumes=consumes,
-            regularity=regularity,
-            name=consumer.name,
-            user_defined_category=define_time_model_for_period(
-                consumer.category or category, target_period=target_period
-            ),
-            fuel=fuel,
-            energy_usage_model={
-                timestep: resolve_and_validate_reference(
-                    value=reference,
-                    references=references.models,
-                )
-                for timestep, reference in define_time_model_for_period(
-                    consumer.energy_usage_model, target_period=target_period
-                ).items()
-            },
-        )
-    else:
-        raise ValueError("Unknown consumer type")
-
-
-TYamlConsumer = TypeVar("TYamlConsumer", bound=Union[YamlCompressor, YamlPump])
+TYamlConsumer = TypeVar("TYamlConsumer", bound=Union[YamlCompressor, YamlPump, YamlTrain[YamlCompressor]])
```

Comment:
In theory you can also make a YamlTrain[YamlPump]? Even though I guess we already agreed that we have no pump trains...

Reply:
Yes